File: ruby/spec/HelloApp_spec.rb
require "HelloApp"

RSpec.describe HelloApp do
  it "should print hello message" do
    messageText = "Hello Test!"
    message = instance_double("HelloMessage", :text => messageText)

    console = instance_double("HelloConsole")
    expect(console).to receive(:print).once.with(messageText)

    app = HelloApp.new message, console
    app.printHello
  end
end

File: scala/src/test/scala/org/hello/HelloAppSpec.scala
package org.hello

import org.mockito.Mockito.{verify, when}
import org.scalatest.flatspec.AnyFlatSpec
import org.scalatestplus.mockito.MockitoSugar.mock

class HelloAppSpec extends AnyFlatSpec {

  "A HelloApp" should "print hello message" in {
    val messageText = "Hello Test!"
    val message = mock[HelloMessage]
    when(message.text).thenReturn(messageText)

    val console = mock[HelloConsole]

    val app = HelloApp(message, console)
    app.printHello()

    verify(console).print(messageText)
  }
}